state/executor.go

```diff
@@ -337,7 +343,7 @@ func (e *BlockExecutor) execute(ctx context.Context, state types.State, block *t
 		return nil, err
 	}
 
-	finalizeBlockResponse, err := e.proxyApp.FinalizeBlock(context.TODO(), &abci.RequestFinalizeBlock{
+	finalizeBlockResponse, err := e.proxyApp.FinalizeBlock(ctx, &abci.RequestFinalizeBlock{
```

Comment:
I also changed this, it didn't have an impact, but it didn't make sense that we wouldn't pass in the ctx that is being passed to the method. 
If there is a reason to use `context.TODO()` then lmk and we can add a comment explaining.